utilities/cloudharness_utilities/deployment-configuration/helm/templates/auto-database.yaml

```diff
@@ -110,6 +110,121 @@ spec:
         persistentVolumeClaim:
           claimName: {{ .app.harness.database.name | quote }}
 ---
+{{ if eq .app.harness.database.type "postgres" }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: "{{ .app.harness.database.name }}-backup"
+  namespace: {{ .root.Values.namespace }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: "{{ .app.harness.database.name }}-backup"
+  template:
+    metadata:
+      labels:
+        app: "{{ .app.harness.database.name }}-backup"
+    spec:
+      securityContext:
+        runAsUser: 999
+        runAsGroup: 999
+      containers:
+        - name: "{{ .app.harness.database.name }}-backup"
+          imagePullPolicy: IfNotPresent
+          image: prodrigestivill/postgres-backup-local
+          env:
+            - name: POSTGRES_HOST
+              value: {{ .app.harness.database.name | quote }}
+            - name: POSTGRES_DB
+              value: {{ (index .app.harness.database .app.harness.database.type).initialdb | quote }}
+            - name: POSTGRES_USER
+              value: {{ .app.harness.database.user | quote }}
+            - name: POSTGRES_PASSWORD
+              value: {{ .app.harness.database.pass | quote }}
+            - name: SCHEDULE
+              value: {{ .app.harness.database.backup.schedule | quote }}
```

Comment:
Backup schedule would be better to keep as a global thing, so that all databases are in sync by design